hack/test-cmd.sh

```diff
@@ -39,13 +40,27 @@ function start-proxy()
   stop-proxy
 
   kube::log::status "Starting kubectl proxy"
-  # the --www and --www-prefix are just to make something definitely show up for
-  # wait_for_url to see.
-  kubectl proxy -p ${PROXY_PORT} --www=. --www-prefix=/healthz "$@" 1>&2 &
+  retry=0
+  until [ $retry -ge 3 ]
+  do
+    set-random-port
+    # the --www and --www-prefix are just to make something definitely show up for
+    # wait_for_url to see.
+    if kubectl proxy -p ${PROXY_PORT} --www=. --www-prefix=/healthz "$@" 1>&2 & then
```

Comment:
Oh, right. Turns out it'll always pass the `if` and break the `do` even when `kubectl proxy` returns error. Change the code to "fail to bind then exit" without retry.